bookkeeper-server/src/main/java/org/apache/bookkeeper/util/PageCacheUtil.java

```diff
@@ -34,22 +36,39 @@ public final class PageCacheUtil {
 
     private static final int POSIX_FADV_DONTNEED = 4; /* fadvise.h */
 
-    private static boolean fadvisePossible = true;
+    private static boolean fadvisePossible = false;
 
     private static final NativeIO NATIVE_IO;
 
     static {
         NativeIO nativeIO = null;
         try {
+            if (SystemUtils.IS_OS_MAC_OSX) {
```

Comment:
Moving the check logic in NativeIOImpl construction will be better. The NativeIOImpl is strongly bound with lib.